ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncAlgorithm.java

```diff
@@ -61,7 +63,28 @@ public CompletableFuture<Void> executeBackwardsSync(final Void unused) {
   public CompletableFuture<Void> pickNextStep() {
     final Optional<Hash> firstHash = context.getBackwardChain().getFirstHashToAppend();
     if (firstHash.isPresent()) {
-      return executeSyncStep(firstHash.get())
+      final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+      executeSyncStep(firstHash.get())
+          .whenComplete(
+              (result, error) -> {
+                if (error != null) {
+                  if (error instanceof CompletionException
+                      && error.getCause() instanceof MaxRetriesReachedException) { // &&
+                    // context.getEthContext().getEthPeers().peerCount() >
+                    // (context.getEthContext().getEthPeers().getMaxPeers() / 2)
```

Comment:
it probably is wise to at least assert we have peers before we drop hashes, but making a qualitative judgement about the peers, when all we have to go by is a hash, is not straightforward.  

IMO dropping the hash is straightforward and not catastrophic, since we will get a subsequent fcU.  As long as we do not give up either:
* too easily 
* in the absence of peers


I think a simple peer count check with a task retry backoff will suffice :+1:

Reply:
We do know that we have at least one peer, otherwise we would not get the max retries exception.

Reply:
Even if we do remove a hash from the list that is on the chain, the BWS still works.